ModulBot/Commands/CreateStat.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using ModulBot.Attributes;
+using ModulBot.Database.PostgresRepositories;
+using ModulBot.Interfaces;
+using ModulBot.Models;
+using Telegram.Bot.Types;
+using Npgsql;
+
+namespace ModulBot.Commands
+{
+    [NotShowInHelp]
+    [Serializable]
+    public class CreateStat : ITool
+    {
+
+        public string Description { get; set; }
+        public List<string> CommandsName { get; set; }
+        public bool IsAdmin { get; set; }
+        private static int step = 0;
+
+        public async void Run(Message message)
+        {
+            if (message.Text[0] == '/')
+            {
+                message.Text = message.Text.Substring(1);
+            }
+
+            switch (step)
+            {
+                case 0:
+                    if (message.Chat.Id < 0) //в групповых чатах бот не видит обычных сообщений
+                    {
+                        await Bot.BotClient.SendTextMessageAsync(message.Chat.Id, Constants.ChatMessage, Telegram.Bot.Types.Enums.ParseMode.Markdown);
+                    }
+
+                    Bot.IsDialogStart = true;
+                    step++;
+                    await Bot.BotClient.SendTextMessageAsync(message.Chat.Id,
+                        "Введите название");
+
+                    break;
+                case 1:
+                    UserState.statName = message.Text;
+                    step++;
+                    await Bot.BotClient.SendTextMessageAsync(message.Chat.Id,
+                        "Введите сообщение");
+
+                    break;
+                case 2:
+                    UserState.statMessage = message.Text;
+                    step++;
+                    await Bot.BotClient.SendTextMessageAsync(message.Chat.Id,
+                        "Введите SQL-запрос");
+
+                    break;
+                case 3:
+                    using (var conn = new NpgsqlConnection(Startup.GetConnectionString()))
+                    {
+                        conn.Open();
+                        try
+                        {
+                            NpgsqlDataReader dr;
+                            var cmd = new NpgsqlCommand(message.Text, conn);
+                            dr = cmd.ExecuteReader();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable 'dr'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS1854)